utils/nctl/sh/scenarios/itst14.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+
+source "$NCTL"/sh/scenarios/common/itst.sh
+
+# Exit if any of the commands fail.
+set -e
+
+#######################################
+# Runs an integration tests that tries to simulate
+# if a validator node can restart within single era 
+# and not equivocate.
+# Arguments:
+#   `timeout=XXX` timeout (in seconds) when syncing.
+#######################################
+function main() {
+    log "------------------------------------------------------------"
+    log "Starting Scenario: itst14"
+    log "------------------------------------------------------------"
+
+    # 0a. Verify all nodes are in sync
+    check_network_sync
+    log "Sleeping 10s"
+    sleep 10
+    check_network_sync
+    # 0b. Get era
+    STOPPED_ERA=$(check_current_era)
+    # 1. Stop node
+    do_stop_node "5"
+    # 2. Let the node go down
+    log_step "Sleeping for 10s"
+    sleep 10
+    # 3. Restart Node
+    do_read_lfb_hash 1
+    do_start_node "5" "$LFB_HASH"
+    # 4. Verify all nodes are in sync
+    check_network_sync
+    # 5. Verify network chain height sync
+    log "Sleeping 10s"
+    sleep 10
+    check_network_sync
+    # 6. Send deploy to stopped node
+    send_transfers '100'
+    # 7. Verify network finalized deploy
+    assert_node_proposed '5'
+    # 8. Check we are in the same era
+    assert_same_era "$STOPPED_ERA"
+    # 9. Wait an era
+    do_await_era_change
+    # 10. Verify all nodes are in sync
+    check_network_sync
+    # 11. Check for equivication
+    check_node_equivocated '5'
+
+    log "------------------------------------------------------------"
+    log "Scenario itst14 complete"
+    log "------------------------------------------------------------"
+}
+
+function assert_same_era() {
+    local ERA=${1}
+    log_step "Checking if within same era..."
+    if [ "$ERA" == "$(check_current_era)" ]; then
+        log "Still within the era. Continuing..."
+    else
+        log "Error: Era progressed! Exiting..."
+        exit 1
+    fi
+}
+
+function assert_node_proposed() {
+    local NODE_ID=${1}
+    local NODE_PATH=$(get_path_to_node $NODE_ID)
+    local PUBLIC_KEY_HEX=$(get_node_public_key_hex $NODE_ID)
+    log_step "Waiting for a node-$NODE_ID to produce a block..."
+    local OUTPUT=$(tail -f "$NODE_PATH/logs/stdout.log" | grep -m 1 "proposer: PublicKey::Ed25519($PUBLIC_KEY_HEX)")
```

Comment:
Do we have a general time out on the IT tests so that we fail if this doesn't return? How is that signaled to the CI? Is it easy to notice?

Reply:
the drone run would eventually time out, (right now its set to 3hrs to accommodate more tests that come in)

Reply:
Can we wrap it with a timeout? I think the `min_era_duration` should do. After all, we expect the node to propose a block rather shortly.

Reply:
yea we can do it, I'll have to adjust the function a bit. The `tail -f` will hold it open indefinitely.